src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,34 @@
 import React from 'react';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
+import './PostInfo.scss';
+
+import { Post } from '../../types/Post';
+
+type Props = {
+  post: Post;
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => (
+  <div className="PostInfo">
+    <div className="PostInfo__header">
+      <h3 className="PostInfo__title">
+        {post.title}
+      </h3>
+
+      <p>
+        {' Posted by  '}
+        {post.user && (<UserInfo user={post.user} />)}
+      </p>
+    </div>
+
+    <p className="PostInfo__body">
+      {post.body}
+    </p>
+
+    {post.comments.length
+      ? <CommentList comments={post.comments} />
+      : <b data-cy="NoCommentsMessage">No comments yet</b>}
+  </div>
 );
```

Comment:
it would be better to use destructuring here to avoid "posts." repeating

const { title, user, comments, } = post